elastic/kibana
# [UA] Inform user about changes to reindexed data

In this PR I made a few changes in Upgrade Assistant to better inform our users about the changes made to their indices when they reindex from inside the app (see [the issue](https://github.com/elastic/kibana/issues/121220) for an detailed explanation of the problem).

## Changes

* For every reindex operation, the user must now confirm that he is aware of the destructive change that will be done

<img width="510" alt="Screenshot 2021-12-22 at 19 04 51" src="https://user-images.githubusercontent.com/2854616/147269135-8077a5cd-4669-4943-9165-2fc5020a52b1.png">

* I've extended the list of steps to 5 (and in some cases 6)

The fourth step of the reindexing server side does the following: create the alias and point it to the new index, delete the original index and - if there are existing aliases for the original index - point those aliases to the new index.

I have decided to add then 2 steps in the list to be explicit on exactly what is happening with the index.

**With no existing aliases**

![reindexing-2](https://user-images.githubusercontent.com/2854616/147271487-d6c4392c-0417-49cb-9420-b3d045e579b5.gif)


**With existing aliases that need to be updated**

![reindexing-1](https://user-images.githubusercontent.com/2854616/147271358-e9e92299-1bcd-4534-9776-b48ed78c8113.gif)


## Notes on implementation

In order to enhance the UI, the reindex status API now return a `meta` object with information about the current index name, the new reindexed index name and possible aliases currently pointing to the index being reindexed.

## How to test

* Download the [data-6.8.20.zip](https://github.com/elastic/kibana/files/7745963/data-6.8.20.zip) and extract the `data` folder
* Launch `7.17` ES with `yarn es snapshot --license trial`
* Stop ES and start it with the dowloaded data `.es/7.17.0/bin/elasticsearch -E path.data=/<absolute-path-to-data-folder>`
* Reset the security password for the default users: in a new terminal launch `.es/7.17.0/bin/elasticsearch-setup-passwords interactive`. For every user give the default password `changeme`
* Start Kibana and navigate to Upgrade assistant
* Under Elasticsearch there should be a few critical issues where reindexing is required
* Make sure that for all of them the user needs to confirm the new checkbox and that the step progress works for both with and without existing aliases.

Fixes https://github.com/elastic/kibana/issues/121220
